src/flamenco/runtime/fd_blockstore.c

```diff
@@ -306,80 +185,118 @@ fd_blockstore_delete( void * shblockstore ) {
   fd_blockstore_t * blockstore = (fd_blockstore_t *)shblockstore;
 
   if( FD_UNLIKELY( !blockstore ) ) {
-    FD_LOG_WARNING( ( "NULL shblockstore" ) );
-    return NULL;
-  }
-
-  if( FD_UNLIKELY( !fd_ulong_is_aligned( (ulong)blockstore, fd_blockstore_align() ) ) ) {
-    FD_LOG_WARNING( ( "misaligned shblockstore" ) );
+    FD_LOG_WARNING(( "NULL shblockstore" ));
     return NULL;
   }
 
-  fd_wksp_t * wksp = fd_wksp_containing( blockstore );
-  if( FD_UNLIKELY( !wksp ) ) {
-    FD_LOG_WARNING( ( "shblockstore must be part of a workspace" ) );
+  if( FD_UNLIKELY( !fd_ulong_is_aligned((ulong)blockstore, fd_blockstore_align() ) )) {
+    FD_LOG_WARNING(( "misaligned shblockstore" ));
     return NULL;
   }
 
   if( FD_UNLIKELY( blockstore->magic != FD_BLOCKSTORE_MAGIC ) ) {
-    FD_LOG_WARNING( ( "bad magic" ) );
+    FD_LOG_WARNING(( "bad magic" ));
     return NULL;
   }
 
-  /* Free all blocks. */
+  /* Delete all structures. */
 
-  ulong * q = fd_wksp_laddr_fast( wksp, blockstore->slot_deque_gaddr );
-  fd_blockstore_slot_deque_remove_all( q );
-  fd_blockstore_slot_deque_push_tail( q, blockstore->smr );
-  while( !fd_blockstore_slot_deque_empty( q ) ) {
-    ulong curr = fd_blockstore_slot_deque_pop_head( q );
+  blockstore->shred_pool = fd_buf_shred_pool_delete( blockstore->shred_pool );
```

Comment:
why are you overwriting the pointers. you can just use (void) on the result of delete